WordPress/wporg-gp-translation-events
# Set up block theme

- Please review only the final result instead of individual commits as there was a lot of trial and error.
- I would recommend checking out the branch and taking a look at the `themes/wporg-translate-events-2024` directory, which contains the new theme.
- The new theme is where the bulk of the changes are, other changes were just to "fake" that theme as the active theme for the current request.

----

This PR introduces a new block theme to be used for translation events, and potentially in the future for the whole of translate.wordpress.org (`wporg-translate`). For the moment, the theme is contained in this repository, but we can extract it into its own repository later, once/if we decide to use the theme for the rest of `wporg-translate`.

**Before proceeding, please read the new theme's [README](https://github.com/WordPress/wporg-gp-translation-events/blob/new-design-setup/themes/wporg-translate-events-2024/README.md)**, which contains valuable information that is useful to review this PR.

## Summary of changes

- [Document](https://github.com/WordPress/wporg-gp-translation-events/blob/new-design-setup/themes/wporg-translate-events-2024/README.md#context) how themes work at `translate.wordpress.org`
- Remove the `templates/new-design` directory
- Create a a new block theme under `themes/wporg-translate-events-2024`
- Add a mechanism to enable the new theme only for specific pages (call `$this->use_theme()` on the `Route` of the page)
- Implement the block theme as described in [this section of the theme's README](https://github.com/WordPress/wporg-gp-translation-events/blob/new-design-setup/themes/wporg-translate-events-2024/README.md#how-it-works-now-with-this-theme)

## How to review and test

1. Pull `translate-w-org-env` and run the provision script again (it will take a while, probably). This is to get the changes introduced by:
    - https://github.com/WordPress/translate-w-org-env/pull/7
    - https://github.com/WordPress/translate-w-org-env/pull/8, 
2. Make sure the Gutenberg plugin is installed/activated, and if not install/activate it
3. Checkout this branch
4. Set the `TRANSLATION_EVENTS_NEW_DESIGN` env variable to `true`
5. Navigate to the "My Events" page
    - You should see the same as the screenshot below
6. Navigate to other pages and make sure they look and function as they did


## Screen capture

This is how a page looks when using the theme.

<img width="936" alt="Screenshot 2024-07-10 at 15 03 10" src="https://github.com/WordPress/wporg-gp-translation-events/assets/550401/42ad338a-2a44-41b1-8a46-e9de09ef5a46">

## TODO

- [x] https://github.com/WordPress/translate-w-org-env/pull/7
- [x] Fix issue with styles when new design not in use
- [x] Remove `add_css()` function
- [x] Remove styles from active theme
- [x] Remove wporg-gp-customizations styles and scripts
- [x] Include styles from parent theme
- [x] Remove `global-styles-inline-css` in favour of the parent theme styles
- [x] Undeprecate header and footer?
- [x] Create a theme
- [x] Load the custom theme
- [x] Implement the basic theme
- [x] Figure out why styles are broken when the theme is not the active theme (it works correctly when it is the active theme)
- [x] Figure out the correct way to use templates/parts/etc
- [x] Remove breadcrumbs block as it's only there as an example block and is not used
- [x] Turn php parts into patterns
- [x] Implement breadcrumbs
- [x] Align navigation to the right
- [x] Fix issue with parent theme style giving a 404
- [x] https://github.com/WordPress/translate-w-org-env/pull/8
